subsys/shell/shell_dummy.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <shell/shell_dummy.h>
+#include <init.h>
+
+SHELL_DUMMY_DEFINE(shell_transport_dummy);
+SHELL_DEFINE(dummy_shell, "~$ ", &shell_transport_dummy, 10,
+	     SHELL_FLAG_OLF_CRLF);
+
+static int init(const struct shell_transport *transport,
+		const void *config,
+		shell_transport_handler_t evt_handler,
+		void *context)
+{
+	struct shell_dummy *sh_dummy = (struct shell_dummy *)transport->ctx;
+
+	if (sh_dummy->initialized) {
+		return -1;
```

Comment:
What about using -EIVNAL or other proper error code instead of -1?

Reply:
OK